compiler/tests/testSuite.bal

```diff
@@ -46,9 +46,20 @@ function testCompileEU(string path, string kind) returns file:Error|io:Error? {
             else if !err.message().includes("'io:println'") {
                 test:assertFalse(err is err:Semantic, "semantic error on U test" + path);
             }
-            int? lineNumber = compileErrorLineNumber(err);
-            if lineNumber != () && (isE || kind[1] == "e") {
-                test:assertEquals(lineNumber, check errorLine(path), "wrong line number in error " + path);
+            if kind == "e" || kind == "ue" {
+                var expectedErrorLocation = checkpanic expectedErrorLocation(err, path);
+                if !(expectedErrorLocation is ()) {
+                    var [expectedFilename, expectedLineNo] = expectedErrorLocation;
```

Comment:
we could avoid this `if` and extra variable, by `expectedErrorLocation` returning and error instead `()` when it can't find a location. Something like `return error("@error annotation not found");` should work. 

Do we need a `checkpanic` ? wouldn't just `check` work?

Reply:
There are test cases without @error annotaions such as `testSuite/05-narrow/13-e.bal` is this expected or is this an error?

Reply:
It's a bug, it should have `@error` on line 6.

Reply:
How was that passing ? Let me check.

Reply:
Looks like the previous logic didn't to the line check if `@error` wasn't there, but please feel free to add it to any file with `-e.bal` suffix.

Reply:
made #484 to fix this